app/controllers/application_controller.rb

```diff
@@ -33,7 +33,7 @@ def after_sign_in_path_for(user)
   end
 
   rescue_from CanCan::AccessDenied do |exception|
-    redirect_to root_url, alert: exception.message
+    redirect_back(fallback_location: root_path, alert: exception.message)
   end
```

Comment:
🏎 The 'not authorised' method will now show up on the page where the user tried to do something forbidden. We couldn't `redirect :back` before, because it would mess up if there wasn't a referer present.